compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -29,6 +28,152 @@ static LogicalResult setRootConfigForPadPipeline(func::FuncOp entryPointFn,
       IREE::Codegen::DispatchLoweringPassPipeline::None);
 }
 
+// This function is based on the following table pulled from the AIEVec_MatMulOp
+// documentation in mlir-aie/include/aie/Dialect/AIEVec/IR/AIEVecOps.td
+//
+//   lhs                | rhs                | Accumulator
+//  :------------------:|:------------------:|:-----------------:
+//   `vector<4x16xi8>`  | `vector<16x8xi4>`  | `vector<4x8xi32>`
+//   `vector<4x8xi8>`   | `vector<8x8xi8>`   | `vector<4x8xi32>`
+//   `vector<4x4xi16>`  | `vector<4x8xi8>`   | `vector<4x8xi32>`
+//   `vector<4x2xi16>`  | `vector<2x8xi16>`  | `vector<4x8xi32>`
+//   `vector<2x8xi16>`  | `vector<8x8xi8>`   | `vector<2x8xi64>`
+//   `vector<4x8xi16>`  | `vector<8x4xi8>`   | `vector<4x4xi64>`
+//   `vector<2x4xi16>`  | `vector<4x8xi16>`  | `vector<2x8xi64>`
+//   `vector<4x4xi16>`  | `vector<4x4xi16>`  | `vector<4x4xi64>`
+//   `vector<4x2xi32>`  | `vector<2x4xi16>`  | `vector<4x4xi64>`
+//   `vector<4x8xbf16>` | `vector<8x4xbf16>` | `vector<4x4xf32>`
+//
+// An instruction size (m, n, k) is returned for each of the above cases. Cases
+// which are not covered by the table return a failure.
```

Comment:
If I understand correctly, the combination of lhs, rhs and accumulator data types should be in the above look up table , otherwise it will return a failure. Currently, the data types in the test of simple-pack pipeline are all i32, then it will return failure? Why there's no instruction for such case?

Reply:
As far as I know AIE does not have a matmul instruction for i32 x i32 -> i32 or i32 x i32 -> i64. The table only has the datatypes which AIE natively supports. 

This function does indeed return failure if the types are not in the table. In this case a default size (currently n=m=4 and k=8) is selected in getPackedSize.  

BTW the test in samples/pack_pipeline_funcIR.mlir is i8 x i8 -> i32 for which the (pre-PR) value of (m=4 n=k=8) is correct according to this table.